app.rb

```diff
@@ -124,4 +124,21 @@ def get_rental(person_id)
       end
     end
   end
+
+  def user_choice(choice)
+    case choice
+    when 1
+      list_book
```

Comment:
- [ ] Creating a reusable method for getting user input can greatly improve code organization and readability. Here's how you can define a reusable method for getting user input and integrate it into your existing code:

❗ Note that the below example is just to show you how this can be implemented.

```ruby
# In a new module, let's call it UserInputHelper
module UserInputHelper
  def get_user_input(prompt)
    puts prompt
    gets.chomp
  end
end
```

Now, you can include the UserInputHelper module in your App class to use the get_user_input method, and you can also use it in other parts of your application:

```ruby
require_relative 'book'
require_relative 'person'
require_relative 'student'
require_relative 'rental'
require_relative 'teacher'
require_relative 'user_input_helper'  # Include the UserInputHelper module

class App
  include UserInputHelper  # Include the module to use get_user_input method

  # ... (other methods)

  def create_a_person
    is_student = get_user_input('Do you want to add a student (1) or a teacher (2)? [Insert the number]').to_i
    age = get_user_input('Age').to_i
    person_name = get_user_input('Name')

    # ... (existing code)
  end

  def create_a_book
    book_title = get_user_input('Title')
    book_author = get_user_input('Author')

    # ... (existing code)
  end

  def create_a_rental
    # ... (existing code)
  end

  # ... (other methods)
end
```
And you can also use the get_user_input method in other parts of your application, like Main class:

```ruby
class Main
  include UserInputHelper  # Include the module to use get_user_input method

  # ... (existing methods)

  def display_menu
    puts
    puts 'Please choose an option by entering a number:'
    puts '1 - List all books'
    puts '2 - List all people'
    puts '3 - Create a person'
    puts '4 - Create a book'
    puts '5 - Create a rental'
    puts '6 - List all rentals for a given person id'
    puts '7 - Exit'
  end

  # ... (existing methods)
end
```

By using this approach, you've created a reusable method for getting user input that can be easily integrated into different parts of your application. This promotes code reuse and enhances the overall maintainability of your codebase.